RDKShell/RDKShell.cpp

```diff
@@ -7162,6 +7230,200 @@ namespace WPEFramework {
+        }
+
+        bool RDKShell::MemCheckpointRestoreClient::restore(const std::string &callSign)
+        {
+            launchRequestThread(RDKShell::MemCheckpointRestoreClient::MEMCR_RESTORE, callSign);
+            return true; // TODO change to void if nothing to check at this point
+        }
+
+
+        void RDKShell::MemCheckpointRestoreClient::launchRequestThread(
+            RDKShell::MemCheckpointRestoreClient::ServerRequestCode cmd, const std::string &callSign)
+        {
+            std::thread requestsThread =
+                std::thread([=]()
+                {
+                    bool success = false;
+                    std::string errorMsg;
+
+                    std::list<uint32_t> processPids;
+                    FindPid(callSign, processPids);
+                    // only one process with callsign as a name is allowed
+                    if (processPids.size() == 1)
+                    {
+                        pid_t appPid = processPids.front();
+                        LOGINFO("Pid found for %s: %d", callSign.c_str(), appPid);
+
+                        RDKShell::MemCheckpointRestoreClient::ServerRequest req = {
+                            .reqCode = cmd,
+                            .pid = appPid};
+
+                        RDKShell::MemCheckpointRestoreClient::ServerResponse resp;
+
+                        success = sendRcvCmd(req, resp, RDKSHELL_THUNDER_TIMEOUT);
+                        if (!success)
+                        {
+                            errorMsg = std::to_string(static_cast<int>(resp.respCode));
+                        }
+                    }
+                    else
+                    {
+                        LOGWARN("Pid not found or more than one found for %s:  %d", callSign.c_str(), processPids.size());
+                    }
+
+                    JsonObject params;
+                    params["callsign"] = callSign;
+                    params["success"] = success;
+                    if (!success)
+                    {
+                        params["reason"] = errorMsg;
+                    }
+
+                    if (cmd == RDKShell::MemCheckpointRestoreClient::MEMCR_CHECKPOINT)
+                    {
+                        mShell.notify(RDKSHELL_EVENT_ON_CHECKPOINTED, params);
+                    }
+                    else if (cmd == RDKShell::MemCheckpointRestoreClient::MEMCR_RESTORE)
+                    {
+                        mShell.notify(RDKSHELL_EVENT_ON_RESTORED, params);
+                    }
+                });
+            requestsThread.detach();
+        }
+
+        bool RDKShell::MemCheckpointRestoreClient::sendRcvCmd(
+            RDKShell::MemCheckpointRestoreClient::ServerRequest &cmd,
+            RDKShell::MemCheckpointRestoreClient::ServerResponse &resp,
+            uint32_t timeouteMs)
+        {
+            const char *serverSocketName = "/tmp/memcrservice";
```

Comment:
Not sure if we should make this socket path configurable?

Reply:
would probably be better if you use a #define  for the socket name.